Updated Sovrn Bid Adaptor for MultiSized and added Error Call Home.

jrosendahl opened the pull request:
<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->
Updated Sovrn Adapter to support multisized ads and record errors

- contact email of the adapter’s maintainer
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

mike-chowla commented on modules/sovrnBidAdapter.js:
```diff
@@ -72,48 +85,83 @@ export const spec = {
+    this.error.data = data
+    this.error.source = 'sovrnBidAdapter'
+    this.error.stack = e.stack
+    this.error.prebidVersion = $$PREBID_GLOBAL$$.version
```
Use of $$PREBID_GLOBAL$$ is not allowed.  Check the other other adapters as to how to get the version substituted in at build time.

jrosendahl replied:
addressed in commit

mike-chowla commented on modules/sovrnBidAdapter.js:
```diff
@@ -72,48 +85,83 @@ export const spec = {
+    this.error.userAgent = navigator.userAgent
+  }
+  send() {
+    ajax(
```
Adapters are not permitted to make their own network requests.  Generally what we suggest is sending the data via getUserSyncs

jrosendahl replied:
addressed in commit

jrosendahl commented:
I have addressed the commit data.  Mike if you are too busy to do this review, would it help if I split multisized support and error tracking into two PR's?

mkendall07 commented:
This pull request **introduces 1 alert** when merging f165cb0c11b2eb28c3c0e98ac51011e19d592ea9 into 2c5685cb271c364478113546cef471c11499ec06 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-4b400237ea00ea80c78e9dbd8ffae65b5fe263d7)

**new alerts:**

* 1 for Missing &#39;this&#39; qualifier

---

*Comment posted by [LGTM.com](https://lgtm.com)*

jrosendahl commented:
So, is there something I need to do to get this merged?  

Please let me know if there is anything I can do to help move this forward.

jsnellbaker commented:
@jrosendahl Can you rebase off of master?  There were some updates made to the test directory and various test files that should address this error.

jrosendahl commented:
Coverage is now at 91% for module.
